bundles/org.openhab.binding.onecta/src/main/java/org/openhab/binding/onecta/internal/api/OnectaConnectionClient.java

```diff
@@ -0,0 +1,355 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.onecta.internal.api;
+
+import static org.openhab.binding.onecta.internal.api.OnectaProperties.*;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.jetty.client.HttpContentResponse;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.onecta.internal.OnectaConfiguration;
+import org.openhab.binding.onecta.internal.api.dto.commands.CommandOnOf;
+import org.openhab.binding.onecta.internal.api.dto.commands.CommandTrueFalse;
+import org.openhab.binding.onecta.internal.api.dto.units.Unit;
+import org.openhab.binding.onecta.internal.api.dto.units.Units;
+import org.openhab.binding.onecta.internal.exception.DaikinCommunicationDataException;
+import org.openhab.binding.onecta.internal.exception.DaikinCommunicationException;
+import org.openhab.binding.onecta.internal.exception.DaikinCommunicationForbiddenException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.*;
+
+/**
+ * @author Alexander Drent - Initial contributionF
+ */
+public class OnectaConnectionClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(OnectaConnectionClient.class);
+    private static final String HTTPHEADER_X_API_KEY = "x-api-key";
+    private static final String HTTPHEADER_BEARER = "Bearer %s";
+    private static final String USER_AGENT_VALUE = "Daikin/1.6.1.4681 CFNetwork/1209 Darwin/20.2.0";
+    private static final String HTTPHEADER_X_API_KEY_VALUE = "xw6gvOtBHq5b1pyceadRp6rujSNSZdjx2AqT03iC";
+    private static final Long REQUEST_TIMEOUT = 60L; // 60 seconds
+
+    private static JsonArray onectaCompleteJsonArrayData = new JsonArray();
+    private static Units onectaUnitsData = new Units();
+    private static OnectaSignInClient onectaSignInClient;
```

Comment:
```suggestion
    private OnectaSignInClient onectaSignInClient = new OnectaSignInClient();
```

why mark it static?

Reply:
I only want 1 connection to Oncta and 1 login. There are multiple places in the binding that need the OnctaConnectionClient